Index: d-i/packages/flash-kernel/README
===================================================================
--- d-i.orig/packages/flash-kernel/README
+++ d-i/packages/flash-kernel/README
@@ -18,6 +18,7 @@
 
  - Linksys NSLU2
  - Thecus N2100
+ - Iomega NAS 100d
 
 If you would like to see support for another device, please file a bug
 report in the Debian Bug Tracking System and include all information
