--- linux-2.6-2.6.18/debian/arch/arm/config.ixp4xx~	2006-12-05 03:50:30.000000000 +1030
+++ linux-2.6-2.6.18/debian/arch/arm/config.ixp4xx	2006-12-05 09:48:28.735778228 +1030
@@ -1735,7 +1735,7 @@
 #
 CONFIG_RTC_DRV_X1205=y
 # CONFIG_RTC_DRV_DS1672 is not set
-# CONFIG_RTC_DRV_PCF8563 is not set
+CONFIG_RTC_DRV_PCF8563=y
 # CONFIG_RTC_DRV_RS5C372 is not set
 # CONFIG_RTC_DRV_M48T86 is not set
 # CONFIG_RTC_DRV_TEST is not set
