---
 debian/arch/arm/config.ixp4xx |    1 +
 1 file changed, 1 insertion(+)

Index: linux-2.6-2.6.20~rc4/debian/arch/arm/config.ixp4xx
===================================================================
--- linux-2.6-2.6.20~rc4.orig/debian/arch/arm/config.ixp4xx
+++ linux-2.6-2.6.20~rc4/debian/arch/arm/config.ixp4xx
@@ -2018,3 +2018,4 @@
 CONFIG_TEXTSEARCH_KMP=m
 CONFIG_TEXTSEARCH_BM=m
 CONFIG_TEXTSEARCH_FSM=m
+
